azure_monitor/src/azure_monitor/sdk/auto_collection/live_metrics/manager.py

```diff
@@ -0,0 +1,170 @@
+                self._ping.shutdown()
+                self._ping = None
+                self._post = LiveMetricsPost(
+                    self._meter, self._instrumentation_key
+                )
+        if self._post:
+            if not self._post.is_user_subscribed:
+                # Switch to Ping
+                self._post.shutdown()
+                self._post = None
+                self._ping = LiveMetricsPing(self._instrumentation_key)
+
+    def shutdown(self):
+        if self._ping:
+            self._ping.shutdown()
+        if self._post:
+            self._post.shutdown()
+        self.thread_event.set()
+
+
+class LiveMetricsPing(threading.Thread):
+    """Ping to Live Metrics service
+
+    Ping to determine if user is subscribed and live metrics need to be send.
+    """
+
+    daemon = True
+
+    def __init__(self, instrumentation_key):
+        super().__init__()
+        self.instrumentation_key = instrumentation_key
+        self.thread_event = threading.Event()
+        self.interval = PING_INTERVAL
+        self.is_user_subscribed = False
+        self.last_send_succeeded = False
+        self.last_request_success_time = 0
+        self.sender = LiveMetricsSender(self.instrumentation_key)
+        self.start()
+
+    def run(self):
+        self.ping()
+        while not self.thread_event.wait(self.interval):
+            self.ping()
+
+    def ping(self):
+        envelope = live_metrics.create_metric_envelope(
+            self.instrumentation_key
+        )
+        token = attach(set_value("suppress_instrumentation", True))
+        response = self.sender.ping(envelope)
+        detach(token)
+        if response.ok:
+            if not self.last_send_succeeded:
+                self.interval = PING_INTERVAL
+                self.last_send_succeeded = True
+            self.last_request_success_time = time.time()
+            if (
+                response.headers.get(
+                    live_metrics.LIVE_METRICS_SUBSCRIBED_HEADER
+                )
+                == "true"
+            ):
+                self.is_user_subscribed = True
+        else:
+            self.last_send_succeeded = False
+            if time.time() >= self.last_request_success_time + 20:
+                self.interval = FALLBACK_INTERVAL
+
+    def shutdown(self):
+        self.thread_event.set()
+
+
+class LiveMetricsPost(threading.Thread):
+    """Post to Live Metrics service
+
+    Post to send live metrics data when user is subscribed.
+    """
+
+    daemon = True
+
+    def __init__(self, meter, instrumentation_key):
+        super().__init__()
+        self.instrumentation_key = instrumentation_key
+        self.meter = meter
+        self.thread_event = threading.Event()
+        self.interval = POST_INTERVAL
+        self.is_user_subscribed = True
+        self.last_send_succeeded = False
+        self.last_request_success_time = time.time()
+        self.exporter = LiveMetricsExporter(self.instrumentation_key)
```

Comment:
The exporter should belong to the manager. You are essentially recreating the exporter everytime a `LiveMetricsPost` is recreated.